app/controllers/changes_controller.rb

```diff
@@ -14,6 +14,6 @@ def undo
       version.item.destroy
       flash[:message] = "#{version.item_type} has been deleted."
     end
-    redirect_back fallback_location: 'public/404.html'
+    redirect_back fallback_location: '/public/404.html'
```

Comment:
This was a bug exposed by the new "open redirect" protection.

In test, at least, this was generating redirects to 'http://test.hostpublic/404.html', which is not the same host as 'test.host' and therefore now raises an exception.